README.md

```diff
@@ -1,40 +1,130 @@
 
-Python 3 Trading Scripts
---------
---------
+Bullish API Examples
+===================
 
-This repository contains Python 3 API trading scripts for [Bullish](https://exchange.bullish.com/) cryptocurrency 
-exchange. This set of scripts can be used out of the box. API traders can use these scripts as a starting point.
+This repository contain code examples to help developers get started with using the [Bullish cryptocurrency exchange](https://exchange.bullish.com/), via a programmatic interface.
 
-Bullish Trading API documentation: https://api.exchange.bullish.com/docs/api/rest/#overview 
+Examples are based on the Bullish API documentation at https://api.exchange.bullish.com/docs/api/rest/#overview 
 
-## Install dependencies
+Getting started
+-------
 
+As with most of this code repository, code snippets below are written in Python 3. [Some Java examples](ecdsa/java-examples/src/main/java/) are also included.
+
+## Connecting to the Bullish API
+For functionality offered via the [Bullish API](https://api.exchange.bullish.com/docs/api/rest/#overview) - some can be accessed publicly (without prior login), and some are private, where a login session is required.
+
+### Example public API - getting supported Markets
+
+```python
+import requests
+import pprint
+
+session = requests.Session()
+
+headers = {
+    "Content-type": "application/json"
+}
+response = session.get(
+    "https://api.exchange.bullish.com/trading-api/v1/markets",
+    headers=headers,
+)
+print(f"Response HTTP Status: {response.status_code}")
+markets = response.json()
+pprint.pp(markets) #pretty print
 ```
-pip3 install -r requirements.txt
+
+### Prerequisite to using private APIs - the JWT token
+Important details about [how to start sending authenticated requests here](https://api.exchange.bullish.com/docs/api/rest/trading-api/v2/#overview--how-to-send-authenticated-requests). Some sample code on how to actually generate a JWT token:
+- ECDSA example: [generate_jwt_v2.py](generate_jwt_v2.py)
+- HMAC example: [generate_jwt_hmac.py](generate_jwt_hmac.py)
```

Comment:
iirc, we wanted to remove these duplicated ones at the top level directory and use the ones inside the respective `/ecdsa`, `/hmac` folders.

Reply:
Same for the other links in this README.

Reply:
Fixed for most, except where the ones in the main folder is more recent. I plan to reorganise all the samples separately, so will update again then